packages/hooks/use-timeout/index.ts

```diff
@@ -0,0 +1,20 @@
+import { onBeforeUnmount } from 'vue'
+
+export default function () {
+  let timeoutHandle: ReturnType<typeof setTimeout>
+
+  onBeforeUnmount(() => {
+    clearTimeout(timeoutHandle)
+  })
+
+  return {
+    registerTimeout: (fn: (...args: any[]) => unknown, delay: number) => {
```

Comment:
Every registerTimeout call will cancel the previous registered call, is this expected?

Reply:
Yes, since each `timeoutHandle` should be co-responding to one specific event.